python/pyspark/ml/tests.py

```diff
@@ -819,6 +847,84 @@ def logistic_regression_check_thresholds(self):
             LogisticRegression, threshold=0.42, thresholds=[0.5, 0.5]
         )
 
+    def test_binomial_logistic_regression_bounds(self):
```

Comment:
Usually it's not need to write exactly the same test as Scala in PySpark, we can use a simple test with loading a dataset  or generating a very simple dataset and run constrained LR on it. You can refer test cases in [test.py](https://github.com/apache/spark/blob/master/python/pyspark/ml/tests.py) or other tests like [this](https://github.com/apache/spark/blob/master/python/pyspark/ml/classification.py#L200).